test/integration/testExecute.js

```diff
@@ -163,180 +165,249 @@ describe('Execute test', function ()
-  {
-    return function (done)
-    {
+  var createItCallback = function (testCase, rowAsserts) {
+    return function (done) {
       {
-        var createFileFormatVariant = `CREATE OR REPLACE FILE FORMAT ${TEST_VARIANT_FORMAT} TYPE = ${testCase.type}`;
-
-        var sampleData;
-        if (testCase.type == 'XML')
-        {
-          sampleData = `<${TEST_HEADER}>${TEST_XML_VAL}</${TEST_HEADER}>`;
-        }
-        else if (testCase.type == 'JSON')
-        {
-          sampleData = `{${TEST_HEADER}: \"${TEST_JSON_VAL}\"}`;
-        }
+        globalConfig.createXmlColumnVariantParserWithParameters({ignoreAttributes: testCase.ignoreAttributes,
+          attributeNamePrefix: testCase.attributeNamePrefix});
 
-        var sampleTempFile = tmp.fileSync({ postfix: testCase.fileExtension });
-        fs.writeFileSync(sampleTempFile.name, sampleData);
+        let sampleTempFile = tmp.fileSync({postfix: testCase.fileExtension});
+        fs.writeFileSync(sampleTempFile.name, testCase.sampleData);
 
         var putVariant = `PUT file://${sampleTempFile.name} @${DATABASE_NAME}.${SCHEMA_NAME}.${TEST_VARIANT_STAGE}`;
 
         // Windows user contains a '~' in the path which causes an error
-        if (process.platform == "win32")
-        {
+        if (process.platform == "win32") {
           var fileName = sampleTempFile.name.substring(sampleTempFile.name.lastIndexOf('\\'));
           putVariant = `PUT file://${process.env.USERPROFILE}\\AppData\\Local\\Temp\\${fileName} @${DATABASE_NAME}.${SCHEMA_NAME}.${TEST_VARIANT_STAGE}`;
         }
 
         async.series(
           [
-            function (callback)
-            {
-              // Create variant table
-              testUtil.executeCmd(connection, createTableVariant, callback);
-            },
-            function (callback)
-            {
-              // Create variant file format
-              testUtil.executeCmd(connection, createFileFormatVariant, callback);
-            },
-            function (callback)
-            {
-              // Create stage with variant file format
-              testUtil.executeCmd(connection, createStageVariant, callback);
-            },
-            function (callback)
-            {
+            function (callback) {
               // Upload sample file
               testUtil.executeCmd(connection, putVariant, callback);
             },
-            function (callback)
-            {
+            function (callback) {
               // Load sample file
               testUtil.executeCmd(connection, copyIntoVariant, callback);
             },
-            function (callback)
-            {
+            function (callback) {
               // Select values from table
               connection.execute({
                 sqlText: selectVariant,
-                complete: function (err, stmt, rows)
-                {
+                complete: function (err, stmt, rows) {
                   var stream = stmt.streamRows();
```

Comment:
when we use streaming let's set `streamResult: true` in parameters or rewrite it to just reading `err` and `rows` without streaming